src/error.rs

```diff
@@ -0,0 +1,125 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::fmt;
+use std::result;
+
+use failure::{Backtrace, Context, Fail};
+use rocket::http::{Status};
+use rocket::response::{Responder, Response};
+use rocket::{self, response, Request};
+use rocket_contrib::Json;
+
+use logging::MozlogLogger;
+
+pub type HandlerResult<T> = result::Result<T, HandlerError>;
+
+#[derive(Debug)]
+pub struct HandlerError {
```

Comment:
(and if we do go back to the name `ApplicationError`, this stuff should probably stay in `app_errors/mod.rs`)

Reply:
I agree with going back to `ApplicationError` now that I think about it, It is is more semantic.

Reply:
I called them HandlerError/Result because ultimately rocket handlers return them as HTTP responses (they impl. rocket's Responder). Though maybe one day [rocket's request guards will also render them as Responders for us (Rocket#596)](https://github.com/SergioBenitez/Rocket/issues/596) -- TBD what that will ultimately look like.

If you really want to change it I'd recommend a shorter "AppError". We'll probably just stick w/ "Handler" in megaphone/pushbox until Rocket#596 shakes out